File: 백준/Silver/1920. 수 찾기/수 찾기.py
def binary_search(x):
    start = 0
    end = N - 1

    while start <= end:
        mid = start + end >> 1

        if arr[mid] == x:
            return 1
        elif arr[mid] > x:
            end = mid - 1
        else:
            start = mid + 1
    return 0


if __name__ == '__main__':
    N = int(input())
    arr = list(map(int, input().split()))
    arr.sort()
    M = int(input())

    for a in list(map(int, input().split())):
        print(binary_search(a))

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int binarySearch(vector<int> &arr, int num)
{
    int left = 0;
    int right = arr.size() - 1;

    while (left <= right)
    {
        int mid = left + (right - left) / 2;

        if (arr[mid] == num)
            return 1;
        else if (arr[mid] > num)
        {
            right = mid - 1;
        }
        else
        {
            left = mid + 1;
        }
    }
    return 0;
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    int N;
    cin >> N;
    vector<int> arr;

    while (N--)
    {
        int num;
        cin >> num;
        arr.push_back(num);
    }

    int M;
    cin >> M;
    sort(arr.begin(), arr.end());
    while (M--)
    {
        int num;
        cin >> num;
        cout << binarySearch(arr, num) << '\n';
    }
}